### ISSUE TITLE
MapperParsingException creating second task

### ISSUE THREAD

**Issue by graphaelli** [elastic/elasticsearch#49802]
**Elasticsearch version**: `Version: 8.0.0-SNAPSHOT, Build: default/docker/4f1ebfaac5c9df1ef5d26f7e036a3d12211c1a0e/2019-12-01T12:32:46.092559Z, JVM: 13.0.1`

**Description of the problem including expected versus actual behavior**:

Unexpected warning logged during task creation.  The operation is still apparently successful.

**Steps to reproduce**:

1. Start elasticsearch 
```
docker run --rm -it -p 9200:9200 -e ES_JAVA_OPTS="-Xms1g -Xmx1g" -e network.host= -e transport.host=127.0.0.1 -e http.host=0.0.0.0 -e xpack.security.enabled=false -e indices.id_field_data.enabled=true docker.elastic.co/elasticsearch/elasticsearch:8.0.0-SNAPSHOT
```
2. Create a task, no warning logged
```
$ curl -H 'Content-Type: application/json' 'http://localhost:9200/anything/_delete_by_query?ignore_unavailable=true&wait_for_completion=false' -d '{"query": {"match_all": {}}}'
{"task":"Qj7Z8Op4Q0GRC0eqNeozNA:131"}
```

es logs:
```
{"type": "server", "timestamp": "2019-12-03T16:59:17,675Z", "level": "INFO", "component": "o.e.t.LoggingTaskListener", "cluster.name": "docker-cluster", "node.name": "25c4c81f44af", "message": "131 finished with response BulkByScrollResponse[took=27.9ms,timed_out=false,sliceId=null,updated=0,created=0,deleted=0,batches=0,versionConflicts=0,noops=0,retries=0,throttledUntil=0s,bulk_failures=[],search_failures=[]]", "cluster.uuid": "gucIWZvTRjmiUzV4WhLT1A", "node.id": "Qj7Z8Op4Q0GRC0eqNeozNA"  }
```

3. Create another task, see warning.  Note that the `delete_by_query` still happens.
```
$ curl -H 'Content-Type: application/json' 'http://localhost:9200/anything/_delete_by_query?ignore_unavailable=true&wait_for_completion=false' -d '{"query": {"match_all": {}}}'
{"task":"Qj7Z8Op4Q0GRC0eqNeozNA:218"}
```

es logs:
<details><summary><code>org.elasticsearch.index.mapper.MapperParsingException: Root mapping definition has unsupported parameters</code></summary>

```
{"type": "server", "timestamp": "2019-12-03T17:05:03,588Z", "level": "DEBUG", "component": "o.e.a.a.i.m.p.TransportPutMappingAction", "cluster.name": "docker-cluster", "node.name": "25c4c81f44af", "message": "failed to put mappings on indices [[[.tasks/r-5kpvC8SkeBjbbJani3jw]]]", "cluster.uuid": "gucIWZvTRjmiUzV4WhLT1A", "node.id": "Qj7Z8Op4Q0GRC0eqNeozNA" , 
"stacktrace": ["org.elasticsearch.index.mapper.MapperParsingException: Root mapping definition has unsupported parameters:  [task : {_meta={version=2}, dynamic=strict, properties={completed={type=boolean}, task={properties={action={type=keyword}, cancellable={type=boolean}, id={type=long}, parent_task_id={type=keyword}, node={type=keyword}, running_time_in_nanos={type=long}, start_time_in_millis={type=long}, type={type=keyword}, status={type=object, enabled=false}, description={type=text}, headers={type=object, enabled=false}}}, response={type=object, enabled=false}, error={type=object, enabled=false}}}]",
"at org.elasticsearch.index.mapper.DocumentMapperParser.checkNoRemainingFields(DocumentMapperParser.java:147) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:135) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:83) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.index.mapper.MapperService.parse(MapperService.java:553) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.metadata.MetaDataMappingService$PutMappingExecutor.applyRequest(MetaDataMappingService.java:255) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.metadata.MetaDataMappingService$PutMappingExecutor.execute(MetaDataMappingService.java:226) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:697) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:319) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:214) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:151) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:629) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]",
"at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]",
"at java.lang.Thread.run(Thread.java:830) [?:?]"] }
{"type": "server", "timestamp": "2019-12-03T17:05:03,591Z", "level": "DEBUG", "component": "o.e.a.s.m.TransportMasterNodeAction", "cluster.name": "docker-cluster", "node.name": "25c4c81f44af", "message": "unexpected exception during publication", "cluster.uuid": "gucIWZvTRjmiUzV4WhLT1A", "node.id": "Qj7Z8Op4Q0GRC0eqNeozNA" , 
"stacktrace": ["org.elasticsearch.index.mapper.MapperParsingException: Root mapping definition has unsupported parameters:  [task : {_meta={version=2}, dynamic=strict, properties={completed={type=boolean}, task={properties={action={type=keyword}, cancellable={type=boolean}, id={type=long}, parent_task_id={type=keyword}, node={type=keyword}, running_time_in_nanos={type=long}, start_time_in_millis={type=long}, type={type=keyword}, status={type=object, enabled=false}, description={type=text}, headers={type=object, enabled=false}}}, response={type=object, enabled=false}, error={type=object, enabled=false}}}]",
"at org.elasticsearch.index.mapper.DocumentMapperParser.checkNoRemainingFields(DocumentMapperParser.java:147) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:135) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:83) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.index.mapper.MapperService.parse(MapperService.java:553) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.metadata.MetaDataMappingService$PutMappingExecutor.applyRequest(MetaDataMappingService.java:255) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.metadata.MetaDataMappingService$PutMappingExecutor.execute(MetaDataMappingService.java:226) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:697) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:319) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:214) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:151) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:629) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]",
"at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]",
"at java.lang.Thread.run(Thread.java:830) [?:?]"] }
{"type": "server", "timestamp": "2019-12-03T17:05:03,596Z", "level": "WARN", "component": "o.e.t.TaskManager", "cluster.name": "docker-cluster", "node.name": "25c4c81f44af", "message": "couldn't store response BulkByScrollResponse[took=1.7ms,timed_out=false,sliceId=null,updated=0,created=0,deleted=0,batches=0,versionConflicts=0,noops=0,retries=0,throttledUntil=0s,bulk_failures=[],search_failures=[]]", "cluster.uuid": "gucIWZvTRjmiUzV4WhLT1A", "node.id": "Qj7Z8Op4Q0GRC0eqNeozNA" , 
"stacktrace": ["org.elasticsearch.index.mapper.MapperParsingException: Root mapping definition has unsupported parameters:  [task : {_meta={version=2}, dynamic=strict, properties={completed={type=boolean}, task={properties={action={type=keyword}, cancellable={type=boolean}, id={type=long}, parent_task_id={type=keyword}, node={type=keyword}, running_time_in_nanos={type=long}, start_time_in_millis={type=long}, type={type=keyword}, status={type=object, enabled=false}, description={type=text}, headers={type=object, enabled=false}}}, response={type=object, enabled=false}, error={type=object, enabled=false}}}]",
"at org.elasticsearch.index.mapper.DocumentMapperParser.checkNoRemainingFields(DocumentMapperParser.java:147) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:135) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:83) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.index.mapper.MapperService.parse(MapperService.java:553) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.metadata.MetaDataMappingService$PutMappingExecutor.applyRequest(MetaDataMappingService.java:255) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.metadata.MetaDataMappingService$PutMappingExecutor.execute(MetaDataMappingService.java:226) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:697) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:319) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:214) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:151) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:629) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]",
"at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]",
"at java.lang.Thread.run(Thread.java:830) [?:?]"] }
{"type": "server", "timestamp": "2019-12-03T17:05:03,601Z", "level": "WARN", "component": "o.e.t.LoggingTaskListener", "cluster.name": "docker-cluster", "node.name": "25c4c81f44af", "message": "218 failed with exception", "cluster.uuid": "gucIWZvTRjmiUzV4WhLT1A", "node.id": "Qj7Z8Op4Q0GRC0eqNeozNA" , 
"stacktrace": ["org.elasticsearch.index.mapper.MapperParsingException: Root mapping definition has unsupported parameters:  [task : {_meta={version=2}, dynamic=strict, properties={completed={type=boolean}, task={properties={action={type=keyword}, cancellable={type=boolean}, id={type=long}, parent_task_id={type=keyword}, node={type=keyword}, running_time_in_nanos={type=long}, start_time_in_millis={type=long}, type={type=keyword}, status={type=object, enabled=false}, description={type=text}, headers={type=object, enabled=false}}}, response={type=object, enabled=false}, error={type=object, enabled=false}}}]",
"at org.elasticsearch.index.mapper.DocumentMapperParser.checkNoRemainingFields(DocumentMapperParser.java:147) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:135) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:83) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.index.mapper.MapperService.parse(MapperService.java:553) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.metadata.MetaDataMappingService$PutMappingExecutor.applyRequest(MetaDataMappingService.java:255) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.metadata.MetaDataMappingService$PutMappingExecutor.execute(MetaDataMappingService.java:226) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:697) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:319) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:214) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:151) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:150) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:188) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:629) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:252) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:215) [elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]",
"at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]",
"at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]",
"at java.lang.Thread.run(Thread.java:830) [?:?]"] }
```

</details>

**Comment by romseygeek** [comment]
This looks related to types removal - the mapping has a root type of `task`, which is no longer permitted.  Do we know where this PUT mapping call is coming from?

**Comment by romseygeek** [comment]
This should be fixed by #50419
